### PR TITLE
lumo: init

### PR DESCRIPTION
**Pull request by jgertm**
###### Motivation for this change
`lumo` is a cool project that gives you an instant ClojureScript REPL.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/interpreters/clojure/lumo.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  version = "1.8.0-beta";
+
+  src = fetchurl {
+    url = "https://github.com/anmonteiro/lumo/releases/download/${version}/lumo_linux64.zip";
+    sha256 = "0p06994w48pbgy8xwc1sz3gg609ardsdhmjafdf7qk4gclyiqs5i";
+  };
+
+  buildInputs = [ unzip ];
+
+  phases = [ "unpackPhase" "installPhase" ];
```
Comment (joachifm) [inline pkgs/development/interpreters/clojure/lumo.nix]: Please do not set phases explicitly, rather disable phases that are causing problems, if any.

File: pkgs/development/interpreters/clojure/lumo.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+stdenv.mkDerivation rec {
+  name = "lumo-${version}";
+  version = "1.8.0-beta";
+
+  src = fetchurl {
+    url = "https://github.com/anmonteiro/lumo/releases/download/${version}/lumo_linux64.zip";
+    sha256 = "0p06994w48pbgy8xwc1sz3gg609ardsdhmjafdf7qk4gclyiqs5i";
+  };
+
+  buildInputs = [ unzip ];
```
Comment (joachifm) [inline pkgs/development/interpreters/clojure/lumo.nix]: `unzip` makes more sense as a `nativeBuildInputs`, as it is presumably used only during the build.

File: pkgs/development/interpreters/clojure/lumo.nix (hunk 1 above)
Comment (jgertm) [reply pkgs/development/interpreters/clojure/lumo.nix]: :shipit:

Comment (jgertm) [reply pkgs/development/interpreters/clojure/lumo.nix]: :+1:

File: pkgs/development/interpreters/clojure/lumo.nix (hunk 2 above)
Comment (jgertm) [reply pkgs/development/interpreters/clojure/lumo.nix]: :+1:

**Comment by joachifm** [comment]
@GrahamcOfBorg build lumo

**Comment by jgertm** [comment]
Just discovered this: 
```bash
> lumo                                                                                                                                                                                                    
lumo: lumo: no version information available (required by lumo)                                                                                                                                                                                               
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: lumo: no version information available (required by lumo)
lumo: symbol lookup error: lumo: undefined symbol: 
```
after removing `pkgs.nodejs-8_x` from my `environment.systemPackages`.

**Comment by andreivolt** [comment]
@jgertm I'm getting the same error and I do have `pkgs.nodejs-8_x` in my system packages

**Comment by hlolli** [comment]
@jgertm and @andreivolt like I mentioned in andreivolt's PR, that lumo 1.8.0 was built and released with node 9.2.0, the 1.8.0-beta was probably in 8_x. But I don't think it should matter.

After trying to build lumo with boot, I can't see a way to prevent those http calls that boot wants to make before starting the build, not to mention the yarn calls and nexe's call to V8 snapshots. So it would probably mean a long list of sha256/url tuplest to download all the classpath dependencies for boot and lumo's dependencies. To me the patchelf sounds good, in that if it works it would be the cleanest way to solve this.

Let me see if it makes sense to use npm2nix and use the entrypoint file lumo.js, it could solve the problems for some distros, as well as seeing if you're useing patchelf correctly aiming at the right libc/libc ++ for the build-id in the nix store. If success, I'll make a PR to your branch.

**Comment by hlolli** [comment]
Thinking out loud: I explicily replaced the path to libstdc++.so.6 (which ldd did not found) and changed the intepreted like was in the PR. It's becoming clear that we are dealing with stripped binary, there's to some, some part is ELF and other part is dynamic, meaning the binary looses its runtime dependencies declerations, so we just have to provide them, guessing we need to provide the same libs as nodejs needs. But that's not my problem at hand tough it will be. The segfault I'm currently receiving idicates that part of the binary was compiled useing really old glibc
```
readelf -sW /nix/store/b891mhri338df0dbpnfxs09jhcq5ir1p-lumo-1.8.0/bin/lumo | grep strchr
   146: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND strchr@GLIBC_2.2.5 (4)
 23803: 0000000001468c20   268 FUNC    GLOBAL DEFAULT   13 u_strchr_60
 28829: 00000000014692b0    94 FUNC    GLOBAL DEFAULT   13 u_strchr32_60
 21409: 0000000001468c20   268 FUNC    GLOBAL DEFAULT   13 u_strchr_60
 30744: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND strchr@@GLIBC_2.2.5
 39405: 00000000014692b0    94 FUNC    GLOBAL DEFAULT   13 u_strchr32_60
```
This undefined `strchr` function seems to be problematic, similar problem has been reported
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=43556
where difference in gcc versions can matter for segfaults.

investigateing further...

**Comment by yegortimoshenko** [comment]
Closing in favor of #44076.

**Pull request closed without merging**